### PR TITLE
Don't schedule dummy tasks

### PR DESCRIPTION
**Pull request by mik-laj**
https://github.com/apache/airflow/issues/7871

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by eladkal** [comment]
I think it's a good idea that `DummyOperator` won't be scheduled but I don't agree with the statement "DummyOperator doesn't actually do work".

For example:
If the Dummy is marked always as success what will happen to cases where `DummyOperator` is used with trigger rules for some flows ?

Example:
```
final = DummyOperator(
                              dag=dag,
                              task_id="error_check",
                              trigger_rule=TriggerRule.ONE_FAILED
                            )
[task1, task2, task3] >> final
```

In this case marking `final` as `FAILED` result in failure of the DAG which invoke `on_failure_callback`.

Also it's possible in the above example that there is ExternalSensor from `dag2` on `final` so it's important that `final` won't be marked as `SUCCESS` automatically.


Probably just need to update `Updating.md` alerting change of behavior as  DummyOperator will not longer honor `trigger_rule`

**Comment by mik-laj** [comment]
But the trigger rules will still be respected. I added this code when the task is to be marked for execution by the scheduler and the trigger rules have already been checked. All callbacks will also be executed, because when the task has callbacks, it will still be passed to the scheduler. 

This optimization applies only to cases when the task has no callbacks and when the decision to execute it has been made.

File: airflow/operators/dummy_operator.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ class DummyOperator(BaseOperator):
     """
     Operator that does literally nothing. It can be used to group tasks in a
     DAG.
+
+    The task that uses this operator is automatically marked as successfully completed by the scheduler
```
Comment (eladkal) [inline airflow/operators/dummy_operator.py]: @mik-laj This is what confused me.
Maybe better saying something like:
The task is evaluated by the scheduler but never processed by the executor
WDYT?

Comment (potiuk) [reply airflow/operators/dummy_operator.py]: Yeah. Might be worth adding!

**Review by potiuk — APPROVED** [review]
Just a small update to description needed but otherwise good.

File: airflow/operators/dummy_operator.py (hunk 1 above)
Comment (mik-laj) [reply airflow/operators/dummy_operator.py]: I updated the description.

**Review by eladkal — APPROVED** [review]

**Pull request merged**